docs/how_to_guides/hosting_validator_models.md

```diff
@@ -0,0 +1,210 @@
+The validator model is referenced in the validator implementation, as well as in the README.
+
+![alt_text](./assets/validator_in_readme.png "image_tooltip")
+
+As we can see, the ToxicLanguage validator currently uses the [Detoxify model](https://github.com/unitaryai/detoxify). In order to use this, we need to install the necessary libraries:
+
+```bash
+# Install FastAPI server dependencies
+pip install fastapi uvicorn
+
+# Option 1 - Install required validator dependencies(General approach - Requires Guardrails)
+guardrails hub install hub://guardrails/toxic_language --install-local-models
+
+# Option 2 - Install only required dependencies (Validator Specific - From README)
+pip install detoxify torch
+```
+
+## Step 3: Wrap the model with FastAPI and run the server
+
+Before implementing the API, we first need to define what the API expects as input and what it will return as output so that the API and server can effectively communicate with each other. 
+
+### The Guardrails Standard Inference Endpoint Schema
+
+Guardrails AI recommends using a similar schema for inputs and outputs across all APIs fronting machine learning models. This approach standardizes the implementation for extracting model inference results out of different models in validators. 
+
+The standard inference request can be found [here](https://github.com/guardrails-ai/guardrails/blob/main/guardrails/validator_base.py#L258). 
+
+The standard input definition looks like this: 
+
+```json
+{
+  "inputs": [
+    {
+      "name": "text",
+      "shape": [1],
+      "data": ["Text we want to check for toxic language"],
+      "datatype": "BYTES"
+    },
+    {
+      "name": "threshold",
+      "shape": [1],
+      "data": [0.5],
+      "datatype": "FP32"
+    }
+  ]
+}
+```
+
+Here, `“text”` contains the text we want to check for toxic language and `“threshold”` contains the confidence threshold for determining toxicity. If the model predicts a toxic level higher than the threshold, then the text is considered toxic. 
+
+The standard output definition looks like this:
+
+```json
+{
+  "model_name": "model_name",
+  "model_version": "1",
+  "outputs": [
+    {
+      "name": "result",
+      "datatype": "BYTES",
+      "shape": [2],
+      "data": ["toxicity", "threat"]
+    }
+  ]
+}
+```
+
+Now it’s time to write our `app.py` to set up the FastAPI server:
+
+```python
+from fastapi import FastAPI, HTTPException
+from pydantic import BaseModel
+from typing import List, Union
+import detoxify
+import torch
+import json
+
+app = FastAPI()
+# Initialize the detoxify model once
+device = "cpu"
+model = detoxify.Detoxify('unbiased-small', device=torch.device(device))
+
+class InputData(BaseModel):
+    name: str
+    shape: List[int]
+    data: Union[List[str], List[float]]
+    datatype: str
+
+class InputRequest(BaseModel):
+    inputs: List[InputData]
+
+class OutputData(BaseModel):
+    name: str
+    datatype: str
+    shape: List[int]
+    data: List[Union[str, List[str]]]
```

Comment:
input data and output data are the same thing. Can we just make them "InferenceData" or something?